primitives/io/src/lib.rs

```diff
@@ -1142,6 +1142,175 @@ pub trait Crypto {
 	}
 }
 
+/// Interfaces for working with elliptic curves related types from within the runtime.
+#[runtime_interface]
+pub trait EllipticCurves {
```

Comment:
Could we add some high level function like [FastAggregateVerify](https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-bls-signature-04#section-3.3.4), a very computational intensive call which is required in eth2 light client sync protocol. We(snowfork) previously do it in runtime [here](https://github.com/Snowfork/snowbridge/blob/2922b973af9d4373a4c7a149988919c35fed9d62/parachain/pallets/ethereum-beacon-client/src/lib.rs#L764-L802) but may not fit into large-scale of validators(e.g: in ethereum mainnet 512 validators in sync committee means at least 342 number of `deserialize_compressed_g1` calls). Will be super helpful if we can provide host function here directly.

Reply:
Being on vacation, I just took a short look at the underlying `milagro_bls` crate: it seems to use Bls12_381 under the hood, so it would be up to you to fork this crate and replace the underlying arithmetic operations by the host function calls we provide. AFAIK we will not provide any high ever function for signature aggregation or verification.

Reply:
@achimcc Many thanks for the reply. Yes, underlying is just the `Bls12_381` elliptic curve and I've heard before will got big boost with recent arkworks improvement so I'm also keen to test with/switch to the arkworks equivalent.

One of my concern is that actually the most weight consuming logic in our implementation [here](https://github.com/Snowfork/snowbridge/blob/3db2d69b40f9b4fe88c869566aedaf56752f4e9b/parachain/pallets/ethereum-beacon-client/src/lib.rs#L776-L779) is to instantiate each `milago` G1 PublicKey in an big raw bytes array(i.e. 342 number of [deserialize_compressed_g1](https://github.com/Snowfork/incubator-milagro-crypto-rust/blob/e8f99cc18ae7105cb4b96e0068b6047198b5342a/src/bls381/core.rs#L209) calls as mentioned above), so my question is except for the arithmetic functions do we also need to wrap some serialize/deserialize host functions declared in [CanonicalSerialize](https://github.com/arkworks-rs/algebra/blob/bc991d44c5e579025b7ed56df3d30267a7b9acac/serialize/src/lib.rs#L86) trait as equivalent? Or with arkworks the G1 point preparation logic can be avoided completely in runtime?

Still knows very little about usage of arkworks api, especially how to use it for ethereum [FastAggregateVerify](https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-bls-signature-04#section-3.3.4) so will be appreciated if you can help to clarify at your convenience.